cmd/exec/cmd.go

```diff
@@ -0,0 +1,157 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exec
+
+import (
+	"context"
+	"fmt"
+
+	contextCMD "github.com/okteto/okteto/cmd/context"
+	"github.com/okteto/okteto/cmd/utils"
+	"github.com/okteto/okteto/pkg/analytics"
+	"github.com/okteto/okteto/pkg/log/io"
+	"github.com/okteto/okteto/pkg/model"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+// execFlags is the input of the user to exec command
+type execFlags struct {
+	manifestPath string
+	namespace    string
+	k8sContext   string
+}
+
+type onExecFinish func()
+
+type metadataTracker interface {
+	SetMetadata(metadata *analytics.TrackExecMetadata)
+	Track()
+}
+
+// Exec executes a command on the remote development container
+type Exec struct {
+	ioCtrl            *io.Controller
+	fs                afero.Fs
+	appRetriever      *appRetriever
+	k8sClientProvider okteto.K8sClientProvider
+	mixpannelTracker  metadataTracker
+
+	getExecutorFunc func(dev *model.Dev, podName string) (executor, error)
+	onExecFinish    []onExecFinish
+}
+
+// NewExec creates a new exec command
+func NewExec(fs afero.Fs, ioCtrl *io.Controller) *Exec {
+	k8sProvider := okteto.NewK8sClientProvider()
+	mixpannelTracker := &mixpannelTrack{
+		trackFunc: analytics.TrackExec,
+	}
+	e := &Exec{
+		ioCtrl:            ioCtrl,
+		fs:                fs,
+		k8sClientProvider: k8sProvider,
+		appRetriever:      newAppRetriever(ioCtrl, k8sProvider),
+		mixpannelTracker:  mixpannelTracker,
+		onExecFinish: []onExecFinish{
+			mixpannelTracker.Track,
+		},
+	}
+	e.getExecutorFunc = e.getExecutor
+	return e
+}
+
+// NewCmdExec returns the cobra exec command
+func (e *Exec) NewCmdExec(ctx context.Context) *cobra.Command {
+	execFlags := &execFlags{}
+
+	cmd := &cobra.Command{
+		Use:                   "exec (DEV SVC) [flags] -- COMMAND [args...]",
+		DisableFlagsInUseLine: true,
+		Short:                 "Execute a command in your development container",
+		Long: `Executes the provided command or the default shell inside a running pod. 
+This command allows you to run tools or perform operations directly on a dev environment that is already running.
+For more information on managing pods, refer to the okteto documentation: https://www.okteto.com/docs/`,
+		Example: `# Run the 'echo this is a test' command inside the pod named 'my-pod'
+okteto exec my-pod -- echo this is a test
+
+# Get an interactive shell session inside the pod named 'my-pod'
+okteto exec my-pod`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			manifestOpts := contextCMD.ManifestOptions{Filename: execFlags.manifestPath, Namespace: execFlags.namespace, K8sContext: execFlags.k8sContext}
+			manifest, err := contextCMD.LoadManifestWithContext(ctx, manifestOpts, e.fs)
```

Comment:
Does this take into consideration the env vars for context and namespace after the refactor done by andrea?

Reply:
Yes, I just tested and its working correctly with the latest changes done